dataprofiler/profilers/profile_builder.py

```diff
@@ -1911,6 +1936,50 @@ def report(self, report_options: dict = None) -> dict:
 
         return _prepare_report(report, output_format, omit_keys)
 
+    @classmethod
+    def load_from_dict(
+        cls,
+        data,
+        options: dict | None = None,
+    ):
+        """
+        Parse attribute from json dictionary into self.
+
+        :param data: dictionary with attributes and values.
+        :type data: dict[string, Any]
+        :param options: options for loading column profiler params from dictionary
+        :type options: Dict | None
+
+        :return: Profiler with attributes populated.
+        :rtype: StructuredProfiler
+        """
+        chi2_matrix = data.pop("chi2_matrix")
+        correlation_matrix = data.pop("correlation_matrix")
+
+        structured_profiler = super().load_from_dict(data)
+
+        structured_profiler.times = defaultdict(float, structured_profiler.times)
+
+        if isinstance(structured_profiler, StructuredProfiler):
+            structured_profiler._col_name_to_idx = {
+                int(k): v for k, v in structured_profiler._col_name_to_idx.items()
+            }
+            structured_profiler._col_name_to_idx = defaultdict(
+                list, structured_profiler._col_name_to_idx
+            )
+            structured_profiler.hashed_row_dict = {
+                int(k): v for k, v in structured_profiler.hashed_row_dict.items()
+            }
+
+        if chi2_matrix is not None:
+            setattr(structured_profiler, "chi2_matrix", np.array(chi2_matrix))
```

Comment:
we seem to have two patterns, one where we setattr and one where we use the profiler.{attr}. Let's choose one.